rep-I0005.rst

```diff
@@ -0,0 +1,164 @@
+::
+
+   REP: I0005
+   Title: Cartesian Motion Extensions for the Simple Message Protocol
+   Authord: Fred Proctor <proctor@nist.gov> and Tre Harrison <william.harrison@nist.gov>
+   Status: Active
+   Type: Process
+   Content-Type: text/x-rst
+   Created: 20-May-2015
+   Post-History: 20-May-2015
+
+Outline
+=======
+
+#. Abstract_
+#. Motivation_
+#. `Message Specifications`_
+#. `Cartesian Move Request`_
+#. `Cartesian Move Reply`_
+#. `Cartesian Move State`_
+#. `Object State`_
+#. References_
+#. `Revision History`_
+#. Copyright_
+
+Abstract
+========
+
+This REP documents extensions to the Simple Message protocol
+that support Cartesian motion, and the reporting of information about
+objects in the robot's workspace.
+
+Motivation
+==========
+
+The Simple Mesage protocol defines a low-level interface to robot
+controllers that can easily be implemented. Together with reference
+implementations of Robot Operating System (ROS) trajectory planners,
+Simple Message provides a way for robot vendors to quickly provide
+baseline ROS functionality to their systems. 
+
+Simple Message is a joint-level interface that requires the external
+system to perform the inverse kinematics calculations. Robot
+controllers typically provide this function internally, and the
+proposed extensions enable external systems to benefit from the
+vendor's built-in kinematics. 
+
+The extensions also include messages for reporting the location and
+other information about objects in the robot's immediate environment
+that may be tracked using built-in capabilities of the robot
+controller. 
+
+Message Specifications
+======================
+
+Cartesian Move Request
+----------------------
+A Cartesian move message requests that the robot move to the specified
+Cartesian position and quaternion orientation along a straight line
```

Comment:
I believe EulerZYX is supported by "all" vendor platforms.  Is anybody aware of an exception?  In the past we have erred on the side of doing heavy lifting on the PC side because the controller side is often limited.  Also, conversion is not trivial (see [here](http://stackoverflow.com/questions/1031005/is-there-an-algorithm-for-converting-quaternion-rotations-to-euler-angle-rotatio))